dart/lib/src/sentry_envelope.dart

```diff
@@ -44,13 +47,18 @@ class SentryEnvelope {
   }
 
   /// Stream binary data representation of `Envelope` file encoded.
-  Stream<List<int>> envelopeStream() async* {
+  Stream<List<int>> envelopeStream(SentryOptions options) async* {
     yield utf8.encode(jsonEncode(
       header.toJson(),
       toEncodable: jsonSerializationFallback,
     ));
     final newLineData = utf8.encode('\n');
     for (final item in items) {
+      if (item.header.type == SentryItemType.attachment) {
+        if (await item.header.length() > options.maxAttachmentSize) {
+          continue;
+        }
+      }
```

Comment:
on Java, the validation happens on `SentryEnvelopeItem.fromAttachment` since the `envelopeStream` is too late, we know in advance if the envelope is bigger or not

Reply:
In Dart we do not know the size in advance :(